pkg/archiver-query.go

```diff
@@ -279,16 +280,120 @@ func IsolateBasicQuery(unparsed string) []string {
     // Non-regex queries can request multiple PVs using this syntax: (PV:NAME:1|PV:NAME:2|...)
     // This function takes queries in this format and breaks them up into a list of individual PVs
     unparsed_clean := strings.TrimSpace(unparsed)
-    if unparsed_clean[0] != '(' || unparsed_clean[len(unparsed_clean)-1] != ')' {
-        // if the statement doesn't have the parentheses, no parsing is necessary
-        return []string{unparsed_clean}
+
+    multiFinder, _ := regexp.Compile(`\([^)]*?\)`)
+    // Identify parenthesis-bound sections
+    multiPhrases := multiFinder.FindAllString(unparsed_clean, -1)
+    // Locate parenthesis-bound sections
+    phraseIdxs := multiFinder.FindAllStringIndex(unparsed, -1)
+
+    // A list of all the possible phrases
+    phraseParts := make([][]string, 0, len(multiPhrases))
+
+    for idx, _ := range multiPhrases {
+        // Strip parenthesis
+        multiPhrases[idx] = strings.Trim(multiPhrases[idx], "()")
+        // Break parsed phrases on "|"
+        phraseParts = append(phraseParts, strings.Split(multiPhrases[idx], "|"))
+    }
+
+    // list of all the configurations for the in-order phrases to be inserted
+    phraseCase := PermuteQuery(phraseParts)
+
+    result := make([]string, 0, len(phraseCase))
+
+    for _, phrase := range phraseCase {
+        createdString := SelectiveInsert(unparsed_clean, phraseIdxs, phrase)
+        result = append(result, createdString)
     }
-    // remove leading and following parentheses
-    unparsed_clean = unparsed_clean[1:len(unparsed_clean)-1]
-    result := strings.Split(unparsed_clean, "|")
+
     return result
 }
 
+func PermuteQuery( inputData [][]string) [][]string {
+    /*
+        Generate all ordered permutations of the input strings to make the following operation occur: 
+
+        input:
+            {
+                {"a", "b"}
+                {"c", "d"}
+            }
+
+        output:
+            {
+                {"a", "c"}
+                {"a", "d"}
+                {"d", "c"}
```

Comment:
This comment should be `{"b", "c"}`, right?

Reply:
Good catch! I've pushed a fix.